include/maths/common/CModel.h

```diff
@@ -245,6 +246,37 @@ class MATHS_COMMON_EXPORT CModelProbabilityParams {
     double m_InitialCountWeight{1.0};
 };
 
+//! \brief Describes the impact of different factors on the anomaly score.
+struct MATHS_COMMON_EXPORT SAnomalyScoreExplanation {
+    enum EAnomalyType { E_UNKNOWN = 0, E_DIP, E_SPIKE };
+
+    EAnomalyType s_AnomalyType{E_UNKNOWN};
+    std::size_t s_AnomalyLength{0};
+    int s_SingleBucketImpact{0};
+    int s_MultiBucketImpact{0};
+    int s_AnomalyCharacteristicsImpact{0};
+    double s_LowerConfidenceBound{std::numeric_limits<double>::infinity()};
+    double s_TypicalValue{std::numeric_limits<double>::infinity()};
+    double s_UpperConfidenceBound{std::numeric_limits<double>::infinity()};
```

Comment:
It feels more natural that these are initialised to quiet_NaN for missing, but don't mind too much if this is a pain to change.